packages/financial-templates-lib/src/price-feed/BigQueryPriceFeed.js

```diff
@@ -0,0 +1,218 @@
+class BigQueryPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs the CryptoWatchPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Function} getTime Returns the current time.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   * @param {Number} decimals Number of decimals to use to convert price to wei.
+   */
+  constructor(logger, web3, lookback, getTime, minTimeBetweenUpdates, decimals = 0) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+    this.lookback = lookback;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+    this.toBN = this.web3.utils.toBN;
+
+    this.convertDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number
+      return this.toBN(parseFixed(number.toString(), decimals).toString());
+    };
+  }
+
+  getCurrentPrice() {
+    return this.currentPrice;
+  }
+
+  async getHistoricalPrice(time) {
+    if (this.lastUpdateTime === undefined) {
+      return undefined;
+    }
+
+    // Using moment package to changeto UTC and create acceptable format for BQ
+    const formattedCurrentTime = moment(time)
+      .utc()
+      .format("YYYY-MM-DD HH:mm:ss");
+
+    // 2592000000 is a month time interval
+    let t2 = new Date(time - 2592000000);
+    t2 = moment(t2)
+      .utc()
+      .format("YYYY-MM-DD HH:mm:ss");
+
+    // Create the query with the needed time interval
+    const query = this.createQuery(t2, formattedCurrentTime);
+
+    // Submit async call to BigQuery
+    let priceResponse;
+    await this.runQuery(query)
+      .then(res => {
+        console.log(res[0].gas_price);
+        priceResponse = res[0].gas_price;
+      })
+      .catch(console.log);
+
+    // Check response.
+    if (!priceResponse) {
+      throw new Error(`🚨Could not parse price result from bigquery: ${priceResponse}`);
+    }
+
+    const returnPrice = this.convertDecimals(priceResponse);
+
+    return returnPrice;
+  }
+
+  getLastUpdateTime() {
+    return this.lastUpdateTime;
+  }
+
+  async update() {
+    // Using current time minus 5 minutes as end-bound of query. This is because the BQ dataset lags ~5 minutes
+    // Using moment package to changeto UTC and create acceptable format for BQ
+    let currentTime = new Date();
+    currentTime = currentTime - 300000;
+    const formattedCurrentTime = moment(currentTime)
+      .utc()
+      .format("YYYY-MM-DD HH:mm:ss");
+
+    // 2592000000 is a month time interval
+    let t2 = new Date(currentTime - 2592000000);
```

Comment:
I agree that using moment data manipulation is much cleaner. I changed the code to use `subtract()`

Reply:
And yes it does force the use of the 1-month price identifier. Since this was the pid that we are using for our contract, I focused on only the 1m, but happy to try and generalize this.